lib/logflare_web/controllers/api/query_controller.ex

```diff
@@ -0,0 +1,39 @@
+defmodule LogflareWeb.Api.QueryController do
+  use LogflareWeb, :controller
+  use OpenApiSpex.ControllerSpecs
+
+  alias Logflare.Endpoints
+  alias LogflareWeb.OpenApi.List
+  alias LogflareWeb.OpenApiSchemas.QueryResult
+
+  action_fallback(LogflareWeb.Api.FallbackController)
+
+  tags(["management"])
+
+  operation(:query,
+    summary: "Execute a query",
+    responses: %{200 => List.response(QueryResult)}
+  )
+
+  def query(%{assigns: %{user: user}} = conn, %{"pg_sql" => sql}) do
+    with {:ok, %{rows: rows}} <- Endpoints.run_query_string(user, {:pg_sql, sql}) do
+      json(conn, %{result: rows})
+    else
+      {:error, err} ->
+        json(conn, %{errors: err})
+    end
+  end
+
+  def query(conn, %{"sql" => sql}) do
+    query(conn, %{"bq_sql" => sql})
+  end
```

Comment:
I would put that before `pg_sql` clause.

Reply:
Should it all be just `sql` and then pick up the type of sql from the backend?

Reply:
@chasers one could have multiple backends attached to a source. Having different query params for each backend makes the caller explicitly specify which backend to be used, and would be the situation for example when they are ingesting into both pg and bq, but they want to query pg for certain queries and bq for certain queries.

Reply:
moved the fn header up.